pkg/services/rpcsrv/subscription_test.go

```diff
@@ -418,6 +435,85 @@ func TestFilteredSubscriptions(t *testing.T) {
 	}
 }
 
+func TestFilteredMempoolSubscriptions(t *testing.T) {
+	// We can’t fit this into TestFilteredSubscriptions because mempool events
+	// doesn't depend on blocks events.
+	priv0 := testchain.PrivateKeyByID(0)
+	goodSender := priv0.GetScriptHash()
+
+	var cases = map[string]struct {
+		params string
+		check  func(*testing.T, *neorpc.Notification)
+	}{
+		"mempool_event matching type": {
+			params: `["mempool_event", {"type":"added"}]`,
+			check: func(t *testing.T, resp *neorpc.Notification) {
+				require.Equal(t, neorpc.MempoolEventID, resp.Event)
+				rmap := resp.Payload[0].(map[string]any)
+				require.Equal(t, "added", rmap["type"].(string))
+			},
+		},
+		"mempool_event matching sender": {
+			params: `["mempool_event", {"sender":"` + goodSender.StringLE() + `"}]`,
+			check: func(t *testing.T, resp *neorpc.Notification) {
+				require.Equal(t, neorpc.MempoolEventID, resp.Event)
+				rmap := resp.Payload[0].(map[string]any)
+				txMap := rmap["transaction"].(map[string]any)
+				sender := txMap["sender"].(string)
+				require.Equal(t, address.Uint160ToString(goodSender), sender)
+			},
+		},
+		"mempool_event matching signer": {
+			params: `["mempool_event", {"signer":"` + goodSender.StringLE() + `"}]`,
+			check: func(t *testing.T, resp *neorpc.Notification) {
+				require.Equal(t, neorpc.MempoolEventID, resp.Event)
+				rmap := resp.Payload[0].(map[string]any)
+				txMap := rmap["transaction"].(map[string]any)
+				signers := txMap["signers"].([]any)
+				signer0 := signers[0].(map[string]any)
+				signer0acc := signer0["account"].(string)
+				require.Equal(t, "0x"+goodSender.StringLE(), signer0acc)
+			},
+		},
+		"mempool_event matching sender, signer and type": {
+			params: `["mempool_event", {"sender":"` + goodSender.StringLE() + `", "signer":"` + goodSender.StringLE() + `", "type":"added"}]`,
+			check: func(t *testing.T, resp *neorpc.Notification) {
+				require.Equal(t, neorpc.MempoolEventID, resp.Event)
+				rmap := resp.Payload[0].(map[string]any)
+				require.Equal(t, "added", rmap["type"].(string))
+				txMap := rmap["transaction"].(map[string]any)
+				sender := txMap["sender"].(string)
+				require.Equal(t, address.Uint160ToString(goodSender), sender)
+				signers := txMap["signers"].([]any)
+				signer0 := signers[0].(map[string]any)
+				signer0acc := signer0["account"].(string)
```

Comment:
Don't split it into different variables, they are used in a single place only. Also, it's a test and the code is rather simple. Use the same format as notary request event test uses.